### PR TITLE
Copyright and license

### PR DESCRIPTION
**Pull request by christian-weiss**
In general this project claims copyright. Good.

As far as i know one should add a time frame for the copyright notice indicating a range between first and last code change. For this project: 2011 to 2017.

This should be use on user interface (frontend footer), documentation (readme) and source code incl. unit tests. All these elements of a project should have a notice that each element (file) is part of the project incl. a project-level (global) copyright range. Do not put other copyright ranges at individual file-level.

This PR adds missing copyright notices to some files and normalizes copyright ranges (years) to a global project-level standard (developer-facing and user-facing). Applying this PR now makes sence even when 2018 is just a view days away (update to 2018 is a simple string operation after applying  this PR)

### CODE REVIEW

**Comment by andig** [comment]
I‘m a bit confused here. A couple of questions:

  - does the current status create any problems?
  - what is the value of this pr?
  - is there a standard for the changes suggested?
  - does that mean we have to update all files each year?

**Comment by jahir** [comment]
this whole "Copyright xxxx-yyyy" thing is mainly something US-specific (AFAIK, IANAL). For Germany (probably whole Europe) it means nothing at all. So, I don't care much for such changes, but as it cleans  things up a bit, it's probably ok.

**Comment by christian-weiss** [comment]
Yes, the copyright declaration is not required for german developers (i think). Contributors and users could be from other countries. But to claim it helps a bit to clarify it for all people in the world - at least to be aware that legal aspects / needs to be considered.

When adding a copyright notice then do it "right" / at least consistent.

**Comment by andig** [comment]
> I do not expect that this project never encounter a infringement court process ever

It surely won't.

> You can count the copyright year range update also as a marketing instrument.

Ok, but then lets to it "right", i.e. until 2018. After this PR with consistent notices it will be easy to `s//` across the codebase. Please apply to `index.html`, too.

File: test/SensorTest.php [hunk 1]
```diff
@@ -4,6 +4,9 @@
  *
  * @package Test
```
Comment (andig) [inline test/SensorTest.php]: double package declarations should not happen. always enforce same order that introduces the least amount of changes (author, copyright, package, license)?

Comment (andig) [reply test/SensorTest.php]: ...or drop package altogether as it's not used for anything (preferred).

**Comment by christian-weiss** [comment]
index.html is already done.
DocBlock elements are now re-ordered.
@package removed

I noticed that @license ist not consistent.
http://opensource.org/licenses/gpl-license.php
http://www.opensource.org/licenses/gpl-license.php
are now normalized to: https://opensource.org/licenses/gpl-license.php 
(HTTPS, to let the reader trust the content)

This is an improvement but not the final solution. 

The links above are showing an introduction page, not a concrete license version. I guess your want to license the code base as "GPL 3.0 or any later". And for that case i would recommend to use: http://www.gnu.org/licenses/gpl.txt (as done by some of your files). This URL will show always the latest version of GPL. 
If you want to use GPL 3.0 (without "or any later") i recommend to use: https://opensource.org/licenses/GPL-3.0 instead.

The topic of this PR is done. Ready to merge, from my point of view.

**Comment by christian-weiss** [comment]
If you plan to normalize the @license thing we should do it with this PR, because it touches nearly every file.

**Comment by J-A-U** [comment]
With vzlogger we had an attempt to add an OpenSSL exception to the GPL license.
Should we do here too?

**Comment by andig** [comment]
Dont think we need to as its not compiled into binaries.

**Comment by andig** [comment]
@christian-weiss everythign‘s still 2017. I don‘t plan do do this again in 10 days?

**Comment by christian-weiss** [comment]
"ext-PDO" is the same case as printed by "php -m" which is the name that the extension author gives that extension. I think php extension names do not follow a naming convention. As you can see in "pcre", "PDO", "Reflection" and "SimpleXML".

I will do the license normalization if you say what's the desired URL and license sentence. And then this PR is ready to merge from my point of view. 

It absolutely okay for me to apply this PR now and to do the "2018 change" when first change in 2018 arrives with another PR. Number of commits and pushes does not hurt. I would like to play agile, doing incremental coherent iterations and continuous integration without delaying stuff.

All your comments are appreciated.
This quality "comment ping pong" is not very effective due to the fact you can not commit such little changes directly on the source branch (fork+PR). It would be more effective if we would work on the same repo, still doing it in separate branches with PRs.

@andig feel free to delay this PR until first commit in 2018 arrives. You can apply the "2018 change" yourself direct after merge of this PR then.

File: htdocs/js/entities.js [hunk 2]
```diff
@@ -5,7 +5,7 @@
  * @author Justin Otherguy <justin@justinotherguy.org>
  * @author Steffen Vogel <info@steffenvogel.de>
  * @author Andreas Götz <cpuidle@gmx.de>
- * @copyright Copyright (c) 2011, The volkszaehler.org project
+ * @copyright Copyright (c) 2011-2017, The volkszaehler.org project
  * @package default
```
Comment (andig) [inline htdocs/js/entities.js]: remove package?

**Comment by andig** [comment]
> It absolutely okay for me to apply this PR now and to do the "2018 change" when first change in 2018 arrives with another PR.

I really don't have time for this. We either do 2018 now or I will have to abandon this PR from my side.

File: htdocs/js/entities.js (hunk 2 above)
Comment (christian-weiss) [reply htdocs/js/entities.js]: Oh, ok. I focused on PHP files only, because thought, that htdocs/js/*.js are generated javascript files. I will change it.

**Comment by christian-weiss** [comment]
Again. Travis failed. False-positive. New travis build node are buggy, i guess.

**Comment by andig** [comment]
Lets see if 74c6668f66c219244ceaf53f19d7815330018f32 helps with diagnosis

**Comment by christian-weiss** [comment]
74c6668 failed on master. Maybe travis build node is not the problem, but the shared application-bootstrapping of httpd (php-pm)... Wondering that this error is so random. I think the unit tests are always in the same sequence. Are there external dependencies (network requests to external endpoints) involved?

**Comment by andig** [comment]
Evtl ein Problem mit dem Cache. Entgegen meiner Vermutung ist apcu nämlich nicht über Prozesse geshared und damir für den Zweck eigentlich untauglich da sich mehrere VZ Middleware Instanzen dann keine Daten teilen (aua). Das erklärt nur nicht warum die Gruppentests failen, denn Gruppen werden gar nicht im Cache abgelegt. Bin grad etwas ratlos...

**Comment by andig** [comment]
@christian-weiss magst du noch alles auf 2018 setzen? Dann sind wir hier wenigstens fertig.

**Comment by christian-weiss** [comment]
please od not be scared by misc/controller/misc/log_open2300.sh - my IDE removes trailing whitespaces, which makes this file appear in this diff

**Comment by andig** [comment]
LGTM

@justinotherguy Kannst Du bitte bestätigen dass GPL v3 die "richtige" Lizenz ist, wie auch bei GitHub abgegeben? In den Quellen hieß es bisher allgemein GPL?

**Comment by christian-weiss** [comment]
To make it easy: COPYING (github project license) has another format and last few paragraphs are missing in COPYING. To be identical, i will add the last few paragraphs to COPYING and change the URL to the TXT version of that license: https://www.gnu.org/licenses/gpl-3.0.txt

**Comment by andig** [comment]
While waiting for justin one last suggestion- could we also rename (`git mv`) COPYING to LICENCE? Because that's what it is and it also appears to be the de facto GH standard...

**Comment by justinotherguy** [comment]
yes, GPL v3 is the one.

**Comment by andig** [comment]
Just noticed that the license line now breaks at >80 chars, not so nice:

     * @license https://www.gnu.org/licenses/gpl-3.0.txt GNU General Public License version 3

I'll probably change it to:

     * @license https://www.gnu.org/licenses/gpl-3.0.txt

**Pull request merged**